packages/db/src/translate/executions.ts

```diff
@@ -9,10 +9,44 @@ import Logger from '@cumulus/logger';
 import { removeNilProperties, returnNullOrUndefinedOrDate } from '@cumulus/common/util';
 import { ValidationError } from '@cumulus/errors';
 import { constructCollectionId, deconstructCollectionId } from '@cumulus/message/Collections';
-import { PostgresExecution, PostgresExecutionRecord } from '../types/execution';
+import { PostgresExecution, PostgresExecutionRecord, PostgresSearchExecutionRecord } from '../types/execution';
 import { ExecutionPgModel } from '../models/execution';
 import { CollectionPgModel } from '../models/collection';
 import { AsyncOperationPgModel } from '../models/async_operation';
+import { PostgresCollectionRecord } from '../types/collection';
+
+export const translatePostgresExecutionToApiExecutionWithoutDbQuery = ({
+  executionRecord,
+  collectionPgRecord,
+}:{
+  executionRecord: PostgresSearchExecutionRecord,
+  collectionPgRecord: Pick<PostgresCollectionRecord, 'name' | 'version'>,
+}): ApiExecutionRecord => {
+  const postfix = executionRecord.arn.split(':').pop();
+  if (!postfix) {
+    throw new Error(`Execution ARN record ${executionRecord.arn} has an invalid postfix and API cannot generate the required 'name' field`);
+  }
+
+  return {
+    name: postfix,
+    status: executionRecord.status ?? 'unknown',
+    arn: executionRecord.arn,
+    duration: executionRecord.duration ?? 0,
+    error: executionRecord.error ?? {},
+    tasks: executionRecord.tasks ?? {},
+    originalPayload: executionRecord.original_payload ?? {},
+    finalPayload: executionRecord.final_payload ?? {},
+    type: executionRecord.workflow_name ?? '',
+    execution: executionRecord.url ?? '',
+    cumulusVersion: executionRecord.cumulus_version ?? '',
+    asyncOperationId: executionRecord.asyncOperationId ?? '',
+    collectionId: constructCollectionId(collectionPgRecord.name, collectionPgRecord.version),
+    parentArn: executionRecord.parentArn ?? '',
+    createdAt: Number(executionRecord.created_at),
+    updatedAt: Number(executionRecord.updated_at),
+    timestamp: Number(executionRecord.timestamp),
+  };
+};
 
 export const translatePostgresExecutionToApiExecution = async (
```

Comment:
Update translatePostgresExecutionToApiExecution to call translatePostgresExecutionToApiExecutionWithoutDbQuery without changing the functionality.